### PR TITLE
Add missing WellMatcher.hpp includes

### PR DESCRIPTION
**Pull request by akva2**
And move some code to compile unit to avoid including WellMatcher.hpp in eclproblem.hh.

Downstream of https://github.com/OPM/opm-common/pull/3328

### CODE REVIEW

**Comment by akva2** [comment]
jenkins build this please

File: ebos/eclactionhandler.hh [hunk 1]
```diff
@@ -62,6 +63,10 @@ public:
                     double sim_time,
                     const TransFunc& updateTrans);
 
+  //! \brief Evaluates UDQ statements.
+  void evalUDQ(const unsigned episodeIdx,
+               UDQState& udq_state);
+
```
Comment (bska) [inline ebos/eclactionhandler.hh:69]: Can you name this `evalUDQAssignments()` or something along those lines please?  Unless we alter the implementation, the function only runs UDQ "assign" statements and does not touch UDQ "define" statements and that's an important distinction.

**Review by bska — COMMENTED** [review]
Looks good for the most part, but `evalUDQ()` isn't sufficiently descriptive here in my opinion.

File: ebos/eclactionhandler.hh (hunk 1 above)
Comment (akva2) [reply ebos/eclactionhandler.hh]: done.

**Comment by akva2** [comment]
jenkins build this please

**Review by bska — APPROVED** [review]
Thanks a lot for the updates.   I'll merge into master.

**Pull request merged**